### ISSUE TITLE
[LifetimeSafety] False-positive invalidation with lambda capturing a container

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#186996]
```cpp
#include <iostream>
#include <vector>

void foo() {
    std::vector<int> v;
    auto print = [&]() { std::cout << v.size(); };
    v.push_back(0);
    print();
}
```

This is because we currently consider even the reference to `v` (which is captured by the lambda) as invalidated after the `push_back`. We need granular paths to represent interior invalidations.

**Comment by iitianpushkar** [comment]
Hey @usx95 , I am interested in this issue, can u assign this to me ?
